drud/ddev
# Mutagen.io caching/asynchronous update feature, fixes #3046, fixes #2361

## The Problem/Issue/Bug:

DDEV/docker webserver performance and lack of inotify/fsnotify filesystem notifications when using NFS.

* #3046
* #2361 

## How this PR Solves The Problem:

Adds `ddev config --mutagen-enabled` 

## Manual Testing Instructions:

Latest artifacts are at: https://github.com/drud/ddev/actions/runs/1075780196

See [docs](https://github.com/drud/ddev/blob/5513cb1951593a19d5830da149b1053b119f7020/docs/users/performance.md#using-mutagen)

In a project, `ddev config --mutagen-enabled` and `ddev start` and `ddev launch`. Experience it.

## Automated Testing Overview:

There are a couple of new automated tests, but also the whole test suite runs with mutagen enabled on Linux, macOS and Windows.

## Release/Deployment notes:

This feature was sponsored by [Tag1 Consulting](https://tag1.com). Thanks! 


<a href="https://gitpod.io/#https://github.com/drud/ddev/pull/3103"><img src="https://gitpod.io/button/open-in-gitpod.svg"/></a>
